sdk/helper/keysutil/policy.go

```diff
@@ -1596,3 +1510,129 @@ func (p *Policy) getVersionPrefix(ver int) string {
 
 	return prefix
 }
+
+type SymmetricOpts struct {
+	Convergent bool
+	ConvergentVersion int
+	Nonce []byte
+	AdditionalData []byte
+	HMACKey []byte
+
+}
+// Symmetrically encrypt a plaintext given the convergence configuration and appropriate keys
+func (p *Policy) SymmetricEncryptRaw(ver int,  encKey, plaintext []byte, opts SymmetricOpts) ([]byte, error) {
```

Comment:
Here and decrypt- since we are being handed an encryption key, should we validate its length?

Reply:
aes.NewCipher and chacha's New both do this already.